<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>720</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Glass Defect Detector</string>
  </property>
  <property name="styleSheet">
   <string>QMainWindow { background-color: #2C3E50; }
QLabel#titleLabel { color: white; font-size: 20px; font-weight: bold; }
QLabel#cameraLabel { color: #D4A048; font-size: 14px; }
QPushButton#startStopButton {
    background-color: #27AE60;
    color: white;
    border: 2px solid #27AE60;
    border-radius: 6px;
    font-size: 14px;
    font-weight: bold;
    padding: 8px 16px;
    min-width: 100px;
}
QPushButton#startStopButton:hover {
    background-color: #229954;
}
QPushButton#startStopButton:pressed {
    background-color: #1E8449;
}
QPushButton#automaticButton, QPushButton#manualButton {
    background-color: #D4A048;
    color: white;
    border: 2px solid #D4A048;
    border-radius: 6px;
    font-size: 12px;
    font-weight: bold;
    padding: 6px 12px;
    min-width: 80px;
}
QPushButton#automaticButton:hover, QPushButton#manualButton:hover {
    background-color: #E5B759;
}
QPushButton#automaticButton:pressed, QPushButton#manualButton:pressed {
    background-color: #C4941F;
    color: white;
}
QPushButton#automaticButton:disabled, QPushButton#manualButton:disabled {
    background-color: #7F8C8D;
    border: 2px solid #7F8C8D;
    color: #5D6D7B;
}
QPushButton#captureButton {
    background-color: #D4A048;
    color: white;
    border: 2px solid #D4A048;
    border-radius: 6px;
    font-size: 12px;
    font-weight: bold;
    padding: 6px 12px;
    min-width: 80px;
}
QPushButton#captureButton:hover {
    background-color: #E5B759;
}
QPushButton#captureButton:pressed {
    background-color: #C4941F;
    color: white;
}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="styleSheet">
    <string>QWidget#centralWidget { background-color: #2C3E50; }</string>
   </property>
   <layout class="QVBoxLayout" name="mainLayout" stretch="0,1">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QWidget" name="headerWidget">
      <property name="minimumHeight">
       <number>80</number>
      </property>
      <property name="maximumHeight">
       <number>80</number>
      </property>
      <property name="styleSheet">
       <string>QWidget#headerWidget { background-color: #34495E; border-bottom: 2px solid #D4A048; }</string>
      </property>
      <layout class="QHBoxLayout" name="headerLayout">
       <property name="leftMargin">
        <number>20</number>
       </property>
       <property name="topMargin">
        <number>10</number>
       </property>
       <property name="rightMargin">
        <number>20</number>
       </property>
       <property name="bottomMargin">
        <number>10</number>
       </property>
       <item>
        <layout class="QVBoxLayout" name="titleLayout">
         <item>
          <widget class="QLabel" name="titleLabel">
           <property name="text">
            <string>Glass Defect Detector</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="cameraLabel">
           <property name="text">
            <string>CAM-001</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="headerSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="startStopButton">
         <property name="text">
          <string>START</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="contentLayout" stretch="2,1">
      <property name="spacing">
       <number>15</number>
      </property>
      <property name="leftMargin">
       <number>15</number>
      </property>
      <property name="topMargin">
       <number>15</number>
      </property>
      <property name="rightMargin">
       <number>15</number>
      </property>
      <property name="bottomMargin">
       <number>15</number>
      </property>
      <item>
       <layout class="QVBoxLayout" name="detectionContainerLayout" stretch="1,0">
        <item>
         <widget class="QFrame" name="detectionFrame">
          <property name="styleSheet">
           <string>QFrame#detectionFrame { border: 3px solid #D4A048; border-radius: 10px; background-color: #F5F5F5; }</string>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="detectionLayout">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="detectionLabel">
             <property name="minimumHeight">
              <number>500</number>
             </property>
             <property name="text">
              <string>Detection Preview</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
             <property name="styleSheet">
              <string>QLabel#detectionLabel { color: #D4A048; font-size: 18px; font-weight: bold; }</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="controlButtonsFrame">
          <property name="minimumHeight">
           <number>60</number>
          </property>
          <property name="styleSheet">
           <string>QFrame#controlButtonsFrame { background-color: #34495E; border: 2px solid #D4A048; border-radius: 8px; padding: 10px; }</string>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <layout class="QHBoxLayout" name="controlButtonsLayout">
           <property name="spacing">
            <number>10</number>
           </property>
           <property name="leftMargin">
            <number>10</number>
           </property>
           <property name="topMargin">
            <number>5</number>
           </property>
           <property name="rightMargin">
            <number>10</number>
           </property>
           <property name="bottomMargin">
            <number>5</number>
           </property>
           <item>
            <widget class="QPushButton" name="automaticButton">
             <property name="text">
              <string>Automatic</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="manualButton">
             <property name="text">
              <string>Manual</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="captureButton">
             <property name="text">
              <string>Capture</string>
             </property>
             <property name="visible">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="buttonSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QFrame" name="defectFrame">
        <property name="styleSheet">
         <string>QFrame#defectFrame { background-color: #F5F5F5; }</string>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QVBoxLayout" name="defectLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QListWidget" name="defectList">
           <property name="styleSheet">
            <string>QListWidget#defectList { background-color: #F5F5F5; border: 2px solid #D4A048; border-radius: 8px; padding: 10px; color: #D4A048; font-size: 12px; }
QListWidget::item { padding: 12px; margin: 5px 0; background-color: #EFEFEF; border-radius: 6px; border-left: 4px solid #D4A048; color: #D4A048; }
QListWidget::item:hover { background-color: #E8E8E8; }
QListWidget::item:selected { background-color: #DCDCDC; }</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>